Yoast/javascript
# LIN-625 Add special quotation marks to wordBoundaries

## Summary


This PR can be summarized in the following changelog entry:

* [Yoast SEO Free] Fixes a bug where the German passive voice assessment would show an error when the text contained a participle enclosed within curly quotation marks (e.g., `“getan”`).
* [yoastseo] Adds various quotation marks to `wordBoundaries` and `removePunctuation`.

## Relevant technical choices:

* The error occurred because a problem in the `hasHabenSeinException` method in `GermanParticiple.js`. In that method, `getIndices` is used to get the index of the participle. This method makes use of `wordBoundaries.js`. In `wordBoundaries.js`, the curly quotes hadn't been added. The participles were initially matched via the route `german/passiveVoice/getParticiples` > `getWords` > `removePunctuation`. There, the curly brackets were present. However, some other brackets were missing there, too, so I've added them.
* In addition to adding the brackets, I've also made the safety check more stringent, so that it now only looks for indices if a participle has been found.
* Some full text specs now have slightly changed (more accurate) results because the improved recognition of quotation marks. 


## Test instructions

This PR can be tested by following these steps:
* Test in Yoast SEO Free.
* Set the language of your site to German
* Open a new post and paste the sentence: `Das heißt, Entscheidungen werden an der Peripherie von den Personen getroffen, die “mitten im Geschehen” sind.`
* You should get the feedback `Passive voice: 100% of the sentences contain passive voice, which is more than the recommended maximum of 10%. Try to use their active counterparts.`. The passive voice analysis shouldn't show an error.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/QAK-2280
